### DISCUSSION TITLE
how to set different fixed time steps for different time periods

### DISCUSSION THREAD

**Discussion by SunJianhang678** [idaholab/moose#29087] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello, I would like to set different fixed time steps for different time periods in one transient simulation. Can I set it directly in the [Executioner] of the input file?
If I use the TimeStepper method, it seems that I need to get the current time to segment the time, is there a way to get the current time in the Timestepper? I tried to use _t and _current_time/getCurrentTime but no help. Thanks for your help!

**Comment by GiudGiud** [comment, marked as answer]
Hello

Different time step cannot be done directly in the Executioner block. You can only do the "ConstantDT" behavior there.
You can use the TimeStepper or the TimeSteppers block.

> I need to get the current time to segment the time, is there a way to get the current time in the Timestepper? I tried to use _t and _current_time/getCurrentTime but no help. Thanks for your help!

The current time is accessible when coding a time stepper using the `_time` attribute. The time is retrieved from the problem.
```
    _time(_fe_problem.time()),
```

> different fixed time steps for different time period

If you already know the size of the fixed time step and the start and end time of the periods, you could use a FunctionDT with a PiecewiseConstant function
